docs/new/develop/indexes.md

```diff
@@ -0,0 +1,99 @@
+When you execute your Substreams for the first time, you are reading the data stored in the flat files of the Substreams provider. 
+
+To improve the performance, the data accessed by the Substreams is cached, so that the second or the third time that you run the Substreams, you can read from the cache, thus saving time and money. This behavior is illustrated in the following diagram.
+
+<figure><img src="../../.gitbook/assets/develop/cache-substreams-flow.png" width="100%" /></figure>
+
+This data indexing is done implicitly every time you run a Substreams for the first time, but Substreams also allows you to explicitly specify which specific data you want to index for further performance improvement.
+
+## Indexes
+
+Substreams has recently introduced the concept of _indexed modules_. An indexed module is a module that has been pre-cached for some specific data. Let's see it with an example!
```

Comment:
I think "index modules" is clearer than "indexed modules".
We are not creating a module that happens to be indexed... We are creating an index! the module is the definition of the index!